api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java

```diff
@@ -440,464 +539,1036 @@ public void setUp() {
+  @EnumSource(
+      value = WorkspaceAccessLevel.class,
+      names = {"OWNER", "WRITER"})
+  public void createCohortReviewAllowedAccessLevel(WorkspaceAccessLevel workspaceAccessLevel) {
     stubBigQueryCohortCalls();
+    // change access, call and check
+    stubWorkspaceAccessLevel(workspace, workspaceAccessLevel);
 
     CohortReview cohortReview =
         cohortReviewController
             .createCohortReview(
-                WORKSPACE_NAMESPACE,
-                WORKSPACE_NAME,
+                workspace.getNamespace(),
+                workspace.getId(),
                 cohortWithoutReview.getCohortId(),
                 cdrVersion.getCdrVersionId(),
                 new CreateReviewRequest().size(1))
             .getBody();
 
-    assertThat(cohortReview.getReviewStatus()).isEqualTo(ReviewStatus.CREATED);
-    assertThat(cohortReview.getCohortName()).isEqualTo(cohortWithoutReview.getName());
-    assertThat(cohortReview.getDescription()).isEqualTo(cohortWithoutReview.getDescription());
-    assertThat(cohortReview.getReviewSize()).isEqualTo(1);
-    assertThat(cohortReview.getParticipantCohortStatuses().size()).isEqualTo(1);
-    assertThat(cohortReview.getParticipantCohortStatuses().get(0).getStatus())
-        .isEqualTo(CohortStatus.NOT_REVIEWED);
+    assertNewlyCreatedCohortReview(cohortReview);
+  }
+
+  @ParameterizedTest(name = "createCohortReviewAllowedAccessLevel WorkspaceAccessLevel={0}")
+  @EnumSource(
+      value = WorkspaceAccessLevel.class,
+      names = {"READER", "NO_ACCESS"})
+  public void createCohortReviewAllowedForbiddenAccessLevel(
+      WorkspaceAccessLevel workspaceAccessLevel) {
+    stubBigQueryCohortCalls();
+    // change access, call and check
+    stubWorkspaceAccessLevel(workspace, workspaceAccessLevel);
+
+    Throwable exception =
+        assertThrows(
+            ForbiddenException.class,
+            () ->
+                cohortReviewController.createCohortReview(
+                    workspace.getNamespace(),
+                    workspace.getId(),
+                    cohortWithoutReview.getCohortId(),
+                    cdrVersion.getCdrVersionId(),
+                    new CreateReviewRequest().size(1)));
+
+    assertForbiddenException(exception);
   }
 
+  ////////// updateCohortReview  //////////
   @Test
-  public void createCohortReviewNoCohortException() {
-    long cohortId = 99;
-    when(workspaceAuthService.getWorkspaceEnforceAccessLevelAndSetCdrVersion(
-            WORKSPACE_NAMESPACE, WORKSPACE_NAME, WorkspaceAccessLevel.WRITER))
-        .thenReturn(workspace);
+  public void updateCohortReviewNoEtag() {
+    stubWorkspaceAccessLevel(workspace2, WorkspaceAccessLevel.WRITER);
 
-    stubBigQueryCohortCalls();
+    CohortReview requestCohortReview =
+        new CohortReview()
+            .cohortReviewId(cohortReview.getCohortReviewId())
+            .cohortId(cohortReview.getCohortId())
+            .etag(null);
```

Comment:
No need to set etag to null, It should be null on creation
```suggestion
    CohortReview requestCohortReview =
        new CohortReview()
            .cohortReviewId(cohortReview.getCohortReviewId())
            .cohortId(cohortReview.getCohortId());
```

Reply:
done